pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -360,3 +332,15 @@ func (m *Matrix) validateParameterInOneOfMatrixOrParams(params []Param) (errs *a
 	}
 	return errs
 }
+
+func (m *Matrix) validateNoWholeArrayResults() (errs *apis.FieldError) {
```

Comment:
can you please add a docstring for this function that links to the issue for whole array results?